features/unmarshal/unmarshal.go

```diff
@@ -193,13 +198,16 @@ func (p *unmarshal) mapField(varName string, field *protogen.Field) {
 		p.P(`if postStringIndex`, varName, ` > l {`)
 		p.P(`return `, p.Ident("io", `ErrUnexpectedEOF`))
 		p.P(`}`)
-		if p.unsafe {
+		switch {
+		case p.unsafe:
 			p.P(`if intStringLen`, varName, ` == 0 {`)
 			p.P(varName, ` = ""`)
 			p.P(`} else {`)
 			p.P(varName, ` = `, p.Ident("unsafe", `String`), `(&dAtA[iNdEx], intStringLen`, varName, `)`)
 			p.P(`}`)
-		} else {
+		case p.unique:
+			p.P(varName, ` = `, "unique.Make[string](string", `(dAtA[iNdEx:postStringIndex`, varName, `])).Value()`)
```

Comment:
Since this is an optimization to reduce memory usage, I think this extra `string` allocation when calling `unique.Make` is not ideal. If you read the implementation of string interning in 1.23, any strings passed to `Make` are cloned to prevent capturing the lifetime of a particularly large string that could have been sliced. Hence, we could use `unsafe.String` here to intern the string and save an extra allocation.

Reply:
Thanks for the review, you are correct. I figured this out in the morning, just haven't had time to update the PR. Seems like unique.Make mskes a copy if needed so unsafe here should be completely fine. Will double check and update the PR on Monday.